pluralsh/plural-artifacts
# Use forked postgres operator with liveness probes and fix backup lifecycle

## Summary
This PR updates the postgres operator to the latest version along with a modification that allows for setting liveness probes for databases. This should fix issues where patroni can't talk to the Kubernetes API anymore. It also uses a newer Spilo image that fixes the backup retention policy which wasn't working. Another small fix is that the operator will now add the necessary annotation for IRSA to service accounts it creates for the postgres pods (so this doesn't need to be done with terraform anymore).

## Test Plan
Local linking. 
- Check that backups are pushed to S3.
- Check that WAL archives are pushed to S3.
- Check that old backups are cleaned up.
- Check that there aren't any errors in the `pg_log` logs (related to send WAL archives to S3)


## Checklist



- [ ]  No images hosted from dockerhub
- [ ]  Are dashboards present to understand the health of the application.  There **must** be at least 1 of these
    - [ ]  all databases should have dashboards
    - [ ]  ideally also have at least cpu/mem utilization dashboards for webserver tier of the app
    - [ ]  you can use `plural from-grafana` to convert a grafana dashboard found via google to our CRD
- [ ]  Are scaling runbooks present
    - [ ]  all databases **must** have scaling runbooks
    - [ ]  you can use the charts in `pluralsh/module-library` to accelerate this
- [ ]  do you need to add config overlays?
    - [ ]  inputing secrets
    - [ ]  configuring autoscaling
- [ ]  If there’s a web-facing component to the app, we need to support OIDC authentication and setting up private networks if no authentication option is viable
- [ ]  All major clouds must be supported
    - [ ]  Azure
    - [ ]  AWS
    - [ ]  GCP
